shim.c

```diff
@@ -388,6 +392,39 @@ static BOOLEAN verify_x509(UINT8 *Cert, UINTN CertSize)
 	return TRUE;
 }
 
+static BOOLEAN verify_eku(UINT8 *Cert, UINTN CertSize)
+{
+	X509 *x509;
+	CONST UINT8 *Temp = Cert;
+	EXTENDED_KEY_USAGE *eku;
+	ASN1_OBJECT *module_signing;
+
+	module_signing = OBJ_nid2obj(OBJ_create("1.3.6.1.4.1.2312.16.1.2",
```

Comment:
One more suggestion - can you make the OID string a #define?  So that this code reads OBJ_create(OID_EKU_MODSIGN, NULL, NULL) or similar?